### ISSUE TITLE
plasma 5.23

### ISSUE THREAD

**Issue by begonia208** [Luwx/LightlyShaders#24]
After my system is updated to plasma 5.23 (neon user edition) I can't find the lightly shaders item in the systemsettings/behaviour of desktop/appearance.
It also doesn't work no more... 
Just to let you know. 
Lighly Shaders is beautiful.

**Comment by TechXero** [comment]
Not building Arch / KDE 5.23.1 ... 
https://pastebin.com/xUgTxPni

**Comment by gianmarcogg03** [comment]
This fork fixed the issue: https://github.com/sr-tream/LightlyShaders

**Comment by begonia208** [comment]
Can you tell where I have to look to customize/activate this fork?
Thank you...

**Comment by TechXero** [comment]
is there a way to PKGBUILD ? Or can I take original's and replace URL in that and build ? I want to add to my own repo for my Distro.

**Comment by ZacharyTalis** [comment]
Oh shoot I forgot to mention, [my ShapeCorners fork](https://github.com/ZacharyTalis/ShapeCorners) now implements a fix (though this new implementation suffers from the Korners bug).

**Comment by ryu-ketsueki** [comment]
> Oh shoot I forgot to mention, [my ShapeCorners fork](https://github.com/ZacharyTalis/ShapeCorners) now implements a fix (though this new implementation suffers from the Korners bug).

@a-parhom 's fork proposed a fix for the Korners bug but it seems it requires some improvement to be a little more resource friendly. I'm using it on my laptop and so far I didn't get any significant slowdowns

**Comment by TechXero** [comment]
I was finally able to build package and added to repo from **[SR-Team Fork](https://github.com/sr-tream/LightlyShaders)** works well enough for now..

**Comment by Luwx** [comment]
After taking a look at the changes that were introduced in 5.23, I honestly don't think it's possible to make this effect work like before. KWin no longer exposes a way to paint only the decoration and shadow underneath the window, and this is key to make the effect look good: painting the shadow at the rounded corner. 
@a-parhom fork works around this issue by shrinking the shadow so that it fills the rounded section, but at the cost of shadows not looking like they were intended to. 
While it's nice to have a more optimized KWin, it's also sad to see these breaking changes and 0 interest of KWin developers to make a native solution to round corners.

**Comment by ryu-ketsueki** [comment]
> After taking a look at the changes that were introduced in 5.23, I honestly don't think it's possible to make this effect work like before. KWin no longer exposes a way to paint only the decoration and shadow underneath the window, and this is key to make the effect look good: painting the shadow at the rounded corner.
> @a-parhom fork works around this issue by shrinking the shadow so that it fills the rounded section, but at the cost of shadows not looking like they were intended to.
> While it's nice to have a more optimized KWin, it's also sad to see these breaking changes and 0 interest of KWin developers to make a native solution to round corners.

How about the one on @ZacharyTalis 's ShapeCorners?

**Comment by begonia208** [comment]
I found https://github.com/matinlotfali/KDE-Rounded-Corners . It really works great.

**Comment by Luwx** [comment]
> > After taking a look at the changes that were introduced in 5.23, I honestly don't think it's possible to make this effect work like before. KWin no longer exposes a way to paint only the decoration and shadow underneath the window, and this is key to make the effect look good: painting the shadow at the rounded corner.
> > @a-parhom fork works around this issue by shrinking the shadow so that it fills the rounded section, but at the cost of shadows not looking like they were intended to.
> > While it's nice to have a more optimized KWin, it's also sad to see these breaking changes and 0 interest of KWin developers to make a native solution to round corners.
> 
> How about the one on @ZacharyTalis 's ShapeCorners?

The other ones I looked also suffer from the same problem, the missing shadow at the corner, it will look like the korner bug. I'll poke the code a bit more, but I'm afraid a solution to this will be very difficult or extremely hacky. 

I'll probably end up archiving this repository.

**Comment by ZacharyTalis** [comment]
Yep, all the forks I know about suffer from the same Korners issue.

You can mitigate Korners somewhat by using the [Inactive Blur](https://store.kde.org/p/1206340/) wallpaper plugin. Here's what my desktop looks like with the plugin installed:

![Tiled KDE desktop featuring a blurred wallpaper.](https://user-images.githubusercontent.com/23324896/142302623-dfd43a27-edf6-4166-bf1a-213f3b159176.png)
